lib/Factory/Transport.php

```diff
@@ -57,9 +60,19 @@ public function create(array $transport)
             $auth['euser'] = Ingo::getUser(false);
         }
 
-        $class = 'Ingo_Transport_' . ucfirst($transport['driver']);
+        // Get transport parameters. 
+        try {
+            $customParams = $coreHooks->callHook('transport_params', 'ingo', [$transportDriver, $transportParams]);
+        } catch (Horde_Exception_HookNotSet $e) {
+            $customParams = null;
+        }
+        if (is_array($customParams)){
+            $transportParams = array_merge($transportParams, $customParams);
+        }
+
+        $class = 'Ingo_Transport_' . ucfirst($transportDriver);
```

Comment:
Please do a follow up PR which would also work on Horde\Ingo\Transport\ as we are transitioning to namespaced-first (and eventually, namespaced-only. For now it's OK though.